### DISCUSSION TITLE
how to build android sdk from source

### DISCUSSION THREAD

**Discussion by Amazonzero** [maplibre/maplibre-native#2291] [Q&A]
When I use source code compilation, I always get an error message when using make Android. Is there a clear documentation to explain this

**Comment by hmtriit** [comment]
hi @Amazonzero 
You changed the CMake version (cmakeVersion) from 3.18.1+ to 3.22.1 in the dependencies.gradle file and tried building again. It fixed the issue, and the build was successful for me.

**Reply by Amazonzero** [reply]
about this 
cd /android
run
BUILDTYPE=Release make apackage

result
①
> Task :MapLibreAndroid:configureCMakeRelWithDebInfo[armeabi-v7a]-2 FAILED
C/C++: CMake Warning (dev) in CMakeLists.txt:
C/C++:   No project() command is present.  The top-level CMakeLists.txt file must
C/C++:   contain a literal, direct call to the project() command.  Add a line of
C/C++:   code such as
C/C++:     project(ProjectName)
C/C++:   near the top of the file, but after cmake_minimum_required().
C/C++:   CMake is pretending there is a "project(Project)" command on the first
C/C++:   line.
C/C++: This warning is for project developers.  Use -Wno-dev to suppress it.
C/C++: CMake Error at /root/maplibre-native-android-v11.0.0/vendor/mapbox-base.cmake:4 (add_subdirectory):
C/C++:   The source directory
C/C++:     /root/maplibre-native-android-v11.0.0/vendor/mapbox-base
C/C++:   does not contain a CMakeLists.txt file.
C/C++: Call Stack (most recent call first):
C/C++:   /root/maplibre-native-android-v11.0.0/CMakeLists.txt:1334 (include)
.....
②
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':MapLibreAndroid:configureCMakeRelWithDebInfo[armeabi-v7a]-2'.
> [CXX1429] error when building with cmake using /root/maplibre-native-android-v11.0.0/platform/android/MapLibreAndroid/src/cpp/CMakeLists.txt: -- The C compiler identification is Clang 14.0.6
  -- The CXX compiler identification is Clang 14.0.6
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /root/android-sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/clang - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /root/android-sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring GL-Native with OpenGL renderer backend
  -- Configuring incomplete, errors occurred!
  See also "/root/maplibre-native-android-v11.0.0/platform/android/MapLibreAndroid/.cxx/RelWithDebInfo/2x6l1g2u/armeabi-v7a/CMakeFiles/CMakeOutput.log".

**Comment by louwers** [comment]
You need to clone the submodules.

https://github.com/maplibre/maplibre-native/blob/main/platform/android/DEVELOPING.md#getting-the-source
